x-pack/plugins/maps/public/connected_components/widget_overlay/layer_control/layer_toc/toc_entry/toc_entry_button/toc_entry_button.tsx

```diff
@@ -0,0 +1,180 @@
+  escapedDisplayName: string;
+  layer: ILayer;
+  onClick: () => void;
+}
+
+type Props = StateProps & OwnProps;
+
+interface State {
+  isFilteredByGlobalTime: boolean;
+}
+
+export class TOCEntryButton extends Component<Props, State> {
+  private _isMounted: boolean = false;
+
+  state: State = {
+    isFilteredByGlobalTime: false,
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._loadIsFilteredByGlobalTime();
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async _loadIsFilteredByGlobalTime() {
+    const isFilteredByGlobalTime = await this.props.layer.isFilteredByGlobalTime();
+    if (this._isMounted) {
+      this.setState({ isFilteredByGlobalTime });
+    }
+  }
+
+  getIconAndTooltipContent(): IconAndTooltipContent {
+    let icon;
+    let tooltipContent = null;
+    const footnotes = [];
+    if (this.props.layer.hasErrors()) {
+      icon = (
+        <EuiIcon
+          aria-label={i18n.translate('xpack.maps.layer.loadWarningAriaLabel', {
+            defaultMessage: 'Load warning',
+          })}
+          size="m"
+          type="alert"
+          color="warning"
+        />
+      );
+      tooltipContent = this.props.layer.getErrors();
+    } else if (this.props.layer.isLayerLoading()) {
+      icon = <EuiLoadingSpinner size="m" />;
+    } else if (!this.props.layer.isVisible()) {
+      icon = <EuiIcon size="m" type="eyeClosed" />;
+      tooltipContent = i18n.translate('xpack.maps.layer.layerHiddenTooltip', {
+        defaultMessage: `Layer is hidden.`,
+      });
+    } else if (!this.props.layer.showAtZoomLevel(this.props.zoom)) {
+      const minZoom = this.props.layer.getMinZoom();
+      const maxZoom = this.props.layer.getMaxZoom();
+      icon = <EuiIcon size="m" type="expand" />;
+      tooltipContent = i18n.translate('xpack.maps.layer.zoomFeedbackTooltip', {
+        defaultMessage: `Layer is visible between zoom levels {minZoom} and {maxZoom}.`,
+        values: { minZoom, maxZoom },
+      });
+    } else {
+      const customIconAndTooltipContent = this.props.layer.getCustomIconAndTooltipContent();
+      if (customIconAndTooltipContent) {
+        icon = customIconAndTooltipContent.icon;
+        if (!customIconAndTooltipContent.areResultsTrimmed) {
+          tooltipContent = customIconAndTooltipContent.tooltipContent;
+        } else {
+          footnotes.push({
+            icon: <EuiIcon color="subdued" type="partial" size="s" />,
+            message: customIconAndTooltipContent.tooltipContent,
+          });
+        }
+      }
+
+      if (this.props.isUsingSearch && this.props.layer.getQueryableIndexPatternIds().length) {
+        footnotes.push({
+          icon: <EuiIcon color="subdued" type="filter" size="s" />,
+          message: i18n.translate('xpack.maps.layer.isUsingSearchMsg', {
+            defaultMessage: 'Results narrowed by search bar',
+          }),
+        });
+      }
+      if (this.state.isFilteredByGlobalTime) {
```

Comment:
as discussed, this can toggle on/off directly when the flag turns on/off.

Reply:
Fixed in https://github.com/elastic/kibana/pull/83006/commits/a97a4cd151a20ed6672baeaa81ae601ed03fb837